functional_testing/zrobot_templates/framework/zos_utils/utils.py

```diff
@@ -38,3 +41,62 @@ def create_container(self, **kwargs):
 
     def create_vm(self, **kwargs):
         return self.handle_blueprint('vm.yaml', **kwargs)
+
+    def check_vnc(self, vnc_ip):
+        vnc = 'vncdotool -s %s' % vnc_ip
+        result,error = self.execute_shell_commands(cmd="%s type %s key enter" % (vnc, repr('ls')))       
+        return error 
+        
+    def enable_ssh_access(self, vnc_ip, username=None, password=None):
+
+        username = username or self.vm_username
+        password = password or self.vm_password
+
+        """
+            Add ssh key to a vm with active vnc protocol.
+        """
+        vnc = 'vncdotool -s %s' % vnc_ip
+        commands = [
+            '%s' % username,
+            '%s' % password,
+            'sudo su',
+            '%s' % password,
+            'sed -i "s/^#PasswordAuthentication yes/PasswordAuthentication yes/g" /etc/ssh/sshd',
+            'sed -i "s/PermitRootLogin yes/PermitRootLogin yes/g" /etc/ssh/sshd',            
+            'service sshd restart'
+        ]
+        for cmd in commands:
+            if "sed" in cmd:
+                self.execute_shell_commands(cmd="%s type %s" % (vnc, repr(cmd)))
+                self.execute_shell_commands(cmd="%s key shift-_ type config key enter" % vnc)
+                time.sleep(1)
+            elif 'https' in cmd:
+                self.execute_shell_commands(cmd="%s type %s" % (vnc, repr(cmd)))
+                self.execute_shell_commands(cmd="%s key shift-:" % vnc)
+            else:
+                self.execute_shell_commands(cmd="%s type %s key enter" % (vnc, repr(cmd)))
+                time.sleep(1)
+
+    def execute_command_inside_vm(self, client, vmip,  cmd, username=None, password=None):
+        username = username or self.vm_username
+        password = password or self.vm_password
+
+        cmd = 'sshpass -p "{password}" ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout=5 {username}@{vmip} "{cmd}"'.format(
+            vmip=vmip,
+            username=username,
+            password=password,
+            cmd=cmd
+        )
+
+        response = client.bash(cmd).get()
+        return response
+    
+    def execute_shell_commands(self, cmd):
+        process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, error = process.communicate()
+        return out.decode('utf-8'), error.decode('utf-8')
+
+    def vm_info(self, vm_name):
```

Comment:
i think it is better to call this function.. get_vm()

Reply:
done